File: src/main/java/com/project/food_delivery/dtos/ProductDataDto.java
package com.project.food_delivery.dtos;

import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
public class ProductDataDto {
    @Schema(description = "Name", example = "Apple")
    private String name;

    @Schema(description = "Product category", example = "fruit")
    private String category;

    @Parameter(required = true, content = @Content(
            mediaType = "application/json",
            schema = @Schema(implementation = ProductCharacteristicDto.class)
    ))
    private ProductCharacteristicDto productCharacteristic;

    @Schema(description = "Place", example = "Metro")
    private String place;

    @Schema(description = "Company", example = "Tasty fruit")
    private String company;
}

File: front-end/src/app/common/product-data-dto.ts
import {ProductCharacteristic} from "./product-characteristic";

export class ProductDataDto {
    public name: string;

    public category: string;

    public productCharacteristic: ProductCharacteristic;

    public place: string;

    public company: string;

    constructor(name: string, category: string, productCharacteristic: ProductCharacteristic,
                place: string, company: string) {
        this.name = name;
        this.category = category;
        this.productCharacteristic = productCharacteristic;
        this.place = place;
        this.company = company;
    }
}
